MapboxSearch/base/src/main/java/com/mapbox/search/base/location/LocationUtils.kt

```diff
@@ -1,16 +1,18 @@
 package com.mapbox.search.base.location
 
-import android.content.Context
-import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.common.location.LocationProvider
+import com.mapbox.common.location.LocationServiceFactory
 import com.mapbox.geojson.BoundingBox
-import com.mapbox.search.base.BaseSearchSdkInitializer
 import kotlin.math.abs
 import kotlin.math.log2
 import kotlin.math.max
 import kotlin.math.min
 
-fun defaultLocationEngine(context: Context = BaseSearchSdkInitializer.app) =
-    LocationEngineProvider.getBestLocationEngine(context)
+fun defaultLocationProvider(): LocationProvider =
+    LocationServiceFactory.getOrCreate()
+        .getDeviceLocationProvider(null)
+        .value
+        ?: throw Exception("Failed to get device location provider")
```

Comment:
This will crash process if location provider is not available. It's better to make `defaultLocationProvider()` return nullable type and assume that location provider might not be available.

Reply:
Fixed